enterprise/internal/licensing/features.go

```diff
@@ -43,7 +43,7 @@ func (f BasicFeature) Check(info *Info) error {
 		}
 		return false
 	}
-	if !info.Plan().HasFeature(featureTrimmed) && !hasFeature(featureTrimmed) {
+	if !info.Plan().HasFeature(featureTrimmed, info.IsExpired()) && !(!info.IsExpired() && hasFeature(featureTrimmed)) {
```

Comment:
To make it more readable, I think we can modify the `hasFeature` function to take the `info.IsExpired()` into account and simplify the condition down here.
```
	// Check if the feature is explicitly allowed via license tag.
	hasFeature := func(want Feature) bool {
		if info.IsExpired() {
			return false;
		}
```


```suggestion
	if !(info.Plan().HasFeature(featureTrimmed, info.IsExpired()) || hasFeature(featureTrimmed)) {
```